parquet/src/main/java/org/apache/iceberg/parquet/ReadConf.java

```diff
@@ -157,6 +164,23 @@ ParquetFileReader reader() {
     return shouldSkip;
   }
 
+  private Map<Long, Long> generateRowGroupsStartRowPos() {
+    ParquetFileReader fileReader = newReader(this.file, ParquetReadOptions.builder().build());
+    Map<Long, Long> offsetToStartRowPosMap = new HashMap<>();
+    long curRowCount = 0;
+    for (int i = 0; i < fileReader.getRowGroups().size(); i += 1) {
+      BlockMetaData meta = fileReader.getRowGroups().get(i);
+      offsetToStartRowPosMap.put(meta.getStartingPos(), curRowCount);
+      curRowCount += meta.getRowCount();
+    }
+
+    return offsetToStartRowPosMap;
+  }
+
+  long[] startRowPositions() {
+    return rowGroupsStartRowPos;
```

Comment:
Can we use the same name for the variable?

Reply:
NP, Done.